public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx

```diff
@@ -48,6 +48,14 @@ export function GeneralSettingsUnconnected({ dashboard, updateTimeZone }: Props)
     dashboard.timepicker.nowDelay = nowDelay;
   };
 
+  const onMaxTimeRangeChange = (maxTimeRange: string) => {
+    dashboard.timepicker.maxTimeRange = maxTimeRange;
+  };
+
+  const onOldestFromChange = (oldestFrom: string) => {
+    dashboard.timepicker.oldestFrom = oldestFrom;
+  };
+
```

Comment:
Do we really need two options? they seem to convey the same information, when we you set these to different time ranges? (say max time range 1y, and oldest from 30d), that feels contradictory

Reply:
I got your point. I made this solution according to https://github.com/grafana/grafana/issues/4288

Reply:
If we restrict the user to go back up to a certain limit and allow the user to select a time range within the maximum interval, then these two options will be very helpful.
But you are absolutely right. So what do you suggest? Should we go for one option or two options?

Reply:
Another thing we can input anyone of these two settings. Both the settings are always not required. So if one can restrict only the time range or only the maximum previous time, they can do it.